### PR TITLE
add testing

### PR DESCRIPTION
**Pull request by Fobhep**

### CODE REVIEW

**Comment by Fobhep** [comment]
@mdellweg: travis says nothing to lint for make lint

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	@echo "  lint         to run flake8 and pylint"
+	@echo "  test-setup   to install dependencies"
+
+
+test-setup:
+	pip install --upgrade pip
+	pip install -r requirements.txt
+
+lint:
+pycodestyle --ignore=E402,E722 --max-line-length=160 . sop
```
Comment (mdellweg) [inline Makefile]: Make syntax needs a tab here

**Review by mdellweg — CHANGES_REQUESTED** [review]
Just one syntax issue.
Overall a great idea.

**Comment by Fobhep** [comment]
should be fine now. errors in travis are due to acutal linting run (to be tackled by #4)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	@echo "  lint         to run flake8 and pylint"
+	@echo "  test-setup   to install dependencies"
+
+
+test-setup:
+	pip install --upgrade pip
+	pip install -r requirements.txt
+
+lint:
+	pycodestyle --ignore=E402,E722 --max-line-length=160 . sop
```
Comment (mdellweg) [inline Makefile]: `sop` is superflous here, since that directory is already covered by recursing in `.`.

Comment (Fobhep) [reply Makefile]: ok. did not know whether pycodestyle is working recursively. thx for pointing out

**Review by mdellweg — APPROVED** [review]
ACK
Maybe, you want to clean up a little. Looking at the  changes, it seems more like 1-2 commits, not 5. ;)

**Comment by Fobhep** [comment]
sure, I wanted to squash it anyways ;)

**Pull request merged**